File: Compra.py
import datetime

class Compra:
    def __init__(self, cliente):
        self.cliente = cliente
        self.productos = []
        self.cantidades = []
        self.total = 0.0

    def agregar_producto(self, producto, cantidad):
        self.productos.append(producto)
        self.cantidades.append(cantidad)
        self.total += producto.precio * cantidad

    def get_total(self):
        return self.total

    def mostrar_detalle_compra(self):
        print(f"Cliente: {self.cliente.get_nombre()}")
        for producto, cantidad in zip(self.productos, self.cantidades):
            print(f"Producto: {producto.get_nombre()} - Cantidad: {cantidad}")
        print(f"Total: ${self.total:.2f}")

    def mostrar_detalle_venta(self):
        detalle = f"Cliente: {self.cliente.get_nombre()}\n"
        for producto, cantidad in zip(self.productos, self.cantidades):
            detalle += f"Producto: {producto.get_nombre()} - Cantidad: {cantidad}\n"
        detalle += f"Total: ${self.total:.2f}\n"
        detalle += f"Fecha de compra: {datetime.date.today()}"
        return detalle

File: Compra.java
import java.util.ArrayList;

public class Compra {
    private Cliente cliente;
    private ArrayList<Producto> productos;
    private ArrayList<Integer> cantidades;
    private double total;

    public Compra(Cliente cliente) {
        this.cliente = cliente;
        this.productos = new ArrayList<>();
        this.cantidades = new ArrayList<>();
        this.total = 0.0;
    }

    public void agregarProducto(Producto producto, int cantidad) {
        productos.add(producto);
        cantidades.add(cantidad);
        total += producto.getPrecio() * cantidad;
    }

    public double getTotal() {
        return total;
    }

    public void mostrarDetalleCompra() {
        System.out.println("Cliente: " + cliente.getNombre());
        for (int i = 0; i < productos.size(); i++) {
            System.out.println("Producto: " + productos.get(i).getNombre() + ", Cantidad: " + cantidades.get(i));
        }
        System.out.println("Total: $" + total);
    }
}
